src/watermarks/watermarks.service.ts

```diff
@@ -1,12 +1,52 @@
-import { Injectable } from '@nestjs/common';
+import { BadRequestException, Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
+import { join } from 'path';
 import { Watermark } from 'src/entity/watermark.entity';
-import { Repository } from 'typeorm';
+import { DeleteResult, Repository } from 'typeorm';
+import { CreateWatermarkDto } from './dto/create-watermark.dto';
+import { UpdateWatermarkDto } from './dto/update-watermark.dto';
+import * as fs from 'fs';
 
 @Injectable()
 export class WatermarksService {
   constructor(
     @InjectRepository(Watermark)
-    private readonly WatermarksRepository: Repository<Watermark>,
+    private readonly watermarksRepository: Repository<Watermark>,
   ) {}
+  async create(dto: CreateWatermarkDto, watermark: Express.Multer.File) {
+    fs.cpSync(
+      watermark.path,
+      join(__dirname, '../../public/watermarks/' + watermark.filename),
+    );
+    return await this.watermarksRepository.save({
+      name: watermark.filename,
+      description: dto.description,
+    });
+  }
+
+  async get(id: number) {
+    const { name, ...watermark } = await this.watermarksRepository.findOneBy({
+      id,
+    });
+    const path = 'http://localhost:3000/watermarks/' + name;
```

Comment:
We don't want to have hardcoded url like this - we should have something like BASE_URL in the env file and use that instead of 'http://localhost:3000